### PR TITLE
Add phishing site detection, redirection and information

### LINKED ISSUE #464: Check for Polkadot phishing sites
... redirect when detected (Metamask does this really well)

### PR DESCRIPTION
**Pull request by Tbaut**
closes #464 

image

Comments / questions
- not happy with the `View` that makes the text shrinked on a 560px centered window. The most straight forward solution I thought about (not great though) was to use a bare html page to render without React, but that's additional styling etc to maintain, so left it as is for now.
- Unlike Metamask, I did not want to add a button to `still allow this website`.

### CODE REVIEW

File: packages/extension-base/src/background/types.ts [hunk 1]
```diff
@@ -351,3 +352,7 @@ export interface ResponseJsonRestore {
 }
 
 export type AllowedPath = typeof ALLOWED_PATH[number];
+
+export interface RequestPhishingRedirect {
+  host: string;
+}
```
Comment (Tbaut) [inline packages/extension-base/src/background/types.ts:358]: stray interface.
```suggestion
```

**Comment by jacogr** [comment]
Ok, the PDF/XML check seems useful. I believe the extension may want to run on those pages? Although the should not try and run? Not sure there is an issue (that I've seen) to be fixed here, but can be wrong.

File: packages/extension-base/src/utils/isXmlOrPdf.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+export default function (pathname: string): boolean {
+  const prohibitedFiles: RegExp[] = [
+    /\.xml$/u,
+    /\.pdf$/u
+  ];
+
+  return prohibitedFiles.some((file) => {
+    console.log(pathname, file, file.test(pathname));
```
Comment (Tbaut) [inline packages/extension-base/src/utils/isXmlOrPdf.ts:11]: stray log
```suggestion
```

File: packages/extension-ui/src/Popup/PhishingDetected.tsx [hunk 3]
```diff
@@ -16,30 +16,28 @@ export default function PhishingDetected (): React.ReactElement<Props> {
...
-          Just because a domain does not appear on this list is not an implicit guarantee of that domain's safety.
-          As always, your transactions are your own responsibility.
+          This redirection is based on a list of websites accessible at https://github.com/polkadot-js/phishing. Note that this is a community-driven, curated list.
+          It might be incomplete or inaccurate.
       `)}
       </Note>
       <Note>
         {t<string>(`
-          If you think this domain is incorrectly flagged or if a blocked legitimate website has resolved its security issues,
+          If you think that this website was flagged incorrectly,
```
Comment (jacogr) [inline packages/extension-ui/src/Popup/PhishingDetected.tsx:35]: For these, use the `<Trans />` component from `i18next`. From memory.

```js
<Trans key='phishing.incorrect'>If you think that this website was flagged incorrectly,
  <IssueLink href='https://github.com/polkadot-js/phishing/issues/new'>please open an issue by clicking here</IssueLink></Trans>
```

The reason is not all languages follow the same structure

**Comment by jacogr** [comment]
The look is fine for now. Not great, but it has the same issue as elsewhere - we actually need a full-screen view at some point.

**Review by jacogr — APPROVED** [review]
lgtm.

**Pull request merged**